src/providers/AppLinkProvider.tsx

```diff
@@ -25,6 +28,48 @@ import {
 
 const APP_LINK_PROTOCOL = 'helium://'
 
+// Define subclasses of Error to return address-specific errors when attempting to process scanned
+// payloads from QR codes
+export enum AddressType {
+  SenderAddress = 'Sender Address',
+  ReceiverAddress = 'Receiver Address',
+}
+export class MissingAddressError extends Error {
+  addressType: AddressType
+
+  constructor(addressType: AddressType) {
+    super(`Missing required ${addressType}`)
+    this.addressType = addressType
+  }
+}
+export class InvalidAddressError extends Error {
+  addressType: AddressType
+
+  constructor(addressType: AddressType) {
+    super(`Invalid ${addressType}`)
+    this.addressType = addressType
+  }
+}
+export class MismatchedAddressError extends Error {
+  addressType: AddressType
+
+  constructor(addressType: AddressType) {
+    super(`Mismatched ${addressType} for account`)
+    this.addressType = addressType
+  }
+}
+const assertValidAddress = (address: string, addressType: AddressType) => {
+  if (!address) throw new MissingAddressError(addressType)
+  if (!Address.isValid(address)) throw new InvalidAddressError(addressType)
+}
+const assertCurrentSenderAddress = async (senderAddress: string) => {
+  const current = await getAddress()
+  const sender = Address.fromB58(`${senderAddress}`)
+  if (!isEqual(current, sender)) {
```

Comment:
instead of doing a deep object compare here, you could just use `current?.b58 !== senderAddress`